### PR TITLE
Add TORCH_CHECK for group < channels for native_channel_shuffle

### LINKED ISSUE #153231: `torch.native_channel_shuffle` crashes with Floating Point Exception when given large integer parameter
## Description
When calling `torch.native_channel_shuffle()` with a large integer parameter on an int32 tensor, the program crashes with a "Floating point exception (core dumped)" error.

## Reproducible code example
```python
import torch

print(torch.__version__)

tensor = torch.tensor([[[[-1217083783, -1180358364,  1566700373],
          [  373925201,   -51186559,  -293506799],
          [ -857066355,   438377754, -2026870899],
          [ 1160214616,   911597557, -1627034189],
          [ -362077815, -2021544643,  1955304279],
          [ -297647964,  1261328359,   708727627],
          [ -223320589,  -860377135,    90163968],
          [ -443149553, -1809033061,   435458131],
          [-1788074486,  1279300504,  -653849396],
          [-2053843853,  -188511294, -1800834678],
          [ 1179696088,   -64674434,  1256467073],
          [-1798628439,  1171736981,  -154044883],
          [-1427341420,  1321994468,   942014428],
          [-1417913084, -1678167428,   731124816],
          [  941378115,  -245163524, -1251581896]],
        ]], dtype=torch.int32)

res = torch.native_channel_shuffle(tensor, 2615494409475630863)
```

## Actual behavior
```
2.7.0+cu126
Floating point exception (core dumped)
```

## Colab
link: https://colab.research.google.com/drive/1V913TehGzpkBDZBFuhNaaOFwrlsHJ_EO?usp=sharing

### Versions

```
Collecting environment information...
PyTorch version: 2.7.0+cu126
Is debug build: False
CUDA used to build PyTorch: 12.6
ROCM used to build PyTorch: N/A

OS: Ubuntu 24.04.1 LTS (x86_64)
GCC version: (Ubuntu 13.3.0-6ubuntu2~24.04) 13.3.0
Clang version: 18.1.8 (++20240731025043+3b5b5c1ec4a3-1~exp1~20240731145144.92)
CMake version: version 4.0.0
Libc version: glibc-2.39

Python version: 3.12.3 (main, Feb  4 2025, 14:48:35) [GCC 13.3.0] (64-bit runtime)
Python platform: Linux-6.8.0-58-generic-x86_64-with-glibc2.39
Is CUDA available: False
CUDA runtime version: 12.8.93
CUDA_MODULE_LOADING set to: N/A
GPU models and configuration: Could not collect
Nvidia driver version: Could not collect
cuDNN version: Could not collect
HIP runtime version: N/A
MIOpen runtime version: N/A
Is XNNPACK available: True

CPU:
Architecture:                         x86_64
CPU op-mode(s):                       32-bit, 64-bit
Address sizes:                        52 bits physical, 57 bits virtual
Byte Order:                           Little Endian
CPU(s):                               384
On-line CPU(s) list:                  0-383
Vendor ID:                            AuthenticAMD
Model name:                           AMD EPYC 9684X 96-Core Processor
CPU family:                           25
Model:                                17
Thread(s) per core:                   2
Core(s) per socket:                   96
Socket(s):                            2
Stepping:                             2
BogoMIPS:                             5099.98
Flags:                                fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush mmx fxsr sse sse2 ht syscall nx mmxext fxsr_opt pdpe1gb rdtscp lm constant_tsc rep_good amd_lbr_v2 nopl nonstop_tsc cpuid extd_apicid aperfmperf rapl pni pclmulqdq monitor ssse3 fma cx16 pcid sse4_1 sse4_2 x2apic movbe popcnt aes xsave avx f16c rdrand lahf_lm cmp_legacy svm extapic cr8_legacy abm sse4a misalignsse 3dnowprefetch osvw ibs skinit wdt tce topoext perfctr_core perfctr_nb bpext perfctr_llc mwaitx cpb cat_l3 cdp_l3 hw_pstate ssbd mba perfmon_v2 ibrs ibpb stibp ibrs_enhanced vmmcall fsgsbase bmi1 avx2 smep bmi2 erms invpcid cqm rdt_a avx512f avx512dq rdseed adx smap avx512ifma clflushopt clwb avx512cd sha_ni avx512bw avx512vl xsaveopt xsavec xgetbv1 xsaves cqm_llc cqm_occup_llc cqm_mbm_total cqm_mbm_local user_shstk avx512_bf16 clzero irperf xsaveerptr rdpru wbnoinvd amd_ppin cppc amd_ibpb_ret arat npt lbrv svm_lock nrip_save tsc_scale vmcb_clean flushbyasid decodeassists pausefilter pfthreshold avic v_vmsave_vmload vgif x2avic v_spec_ctrl vnmi avx512vbmi umip pku ospke avx512_vbmi2 gfni vaes vpclmulqdq avx512_vnni avx512_bitalg avx512_vpopcntdq la57 rdpid overflow_recov succor smca fsrm flush_l1d debug_swap
Virtualization:                       AMD-V
L1d cache:                            6 MiB (192 instances)
L1i cache:                            6 MiB (192 instances)
L2 cache:                             192 MiB (192 instances)
L3 cache:                             2.3 GiB (24 instances)
NUMA node(s):                         2
NUMA node0 CPU(s):                    0-95,192-287
NUMA node1 CPU(s):                    96-191,288-383
Vulnerability Gather data sampling:   Not affected
Vulnerability Itlb multihit:          Not affected
Vulnerability L1tf:                   Not affected
Vulnerability Mds:                    Not affected
Vulnerability Meltdown:               Not affected
Vulnerability Mmio stale data:        Not affected
Vulnerability Reg file data sampling: Not affected
Vulnerability Retbleed:               Not affected
Vulnerability Spec rstack overflow:   Mitigation; Safe RET
Vulnerability Spec store bypass:      Mitigation; Speculative Store Bypass disabled via prctl
Vulnerability Spectre v1:             Mitigation; usercopy/swapgs barriers and __user pointer sanitization
Vulnerability Spectre v2:             Mitigation; Enhanced / Automatic IBRS; IBPB conditional; STIBP always-on; RSB filling; PBRSB-eIBRS Not affected; BHI Not affected
Vulnerability Srbds:                  Not affected
Vulnerability Tsx async abort:        Not affected

Versions of relevant libraries:
[pip3] numpy==2.2.5
[pip3] nvidia-cublas-cu12==12.6.4.1
[pip3] nvidia-cuda-cupti-cu12==12.6.80
[pip3] nvidia-cuda-nvrtc-cu12==12.6.77
[pip3] nvidia-cuda-runtime-cu12==12.6.77
[pip3] nvidia-cudnn-cu12==9.5.1.17
[pip3] nvidia-cufft-cu12==11.3.0.4
[pip3] nvidia-curand-cu12==10.3.7.77
[pip3] nvidia-cusolver-cu12==11.7.1.2
[pip3] nvidia-cusparse-cu12==12.5.4.2
[pip3] nvidia-cusparselt-cu12==0.6.3
[pip3] nvidia-nccl-cu12==2.26.2
[pip3] nvidia-nvjitlink-cu12==12.6.85
[pip3] nvidia-nvtx-cu12==12.6.77
[pip3] optree==0.15.0
[pip3] torch==2.7.0
[pip3] triton==3.3.0
[conda] Could not collect
```

### PR DESCRIPTION
**Pull request by aishwaryar12309**
Fixes #153231 

### Changes
- Added `TORCH_CHECK(groups <= channels)` to prevent silent misbehavior in `native_channel_shuffle` when the number of groups is larger than the input's channel dimension. Files changed: aten/src/ATen/native/ChannelShuffle.cpp

### Motivation
- Previously, the function allowed `groups > channels`, which silently produced incorrect behavior or segmentation faults downstream. This check ensures correctness and surfaces misuse early.

Labels: module: nn, module: crash, module: edge cases
cc @albanD

### CODE REVIEW

**Review by albanD — COMMENTED** [review]
Thanks!
Would you be able to add a small test in test_torch.py to make sure it raises as expected now?

**Comment by aishwaryar12309** [comment]
I can add a test!

**Comment by malfet** [comment]
> Thanks! Would you be able to add a small test in test_torch.py to make sure it raises as expected now?

Or to OpInfo as error_inputs ;)

**Comment by huydhn** [comment]
From the OH, you could add `USE_CUDA=0` to build without CUDA.  The local build is much faster that way

**Comment by abhishek-iitmadras** [comment]
This https://github.com/pytorch/pytorch/wiki/lintrunner help to pass lint failure.
